python/src/opendp/extras/sklearn/linear_model/__init__.py

```diff
@@ -0,0 +1,125 @@
+We suggest importing under the conventional name ``dp``:
+
+.. code:: python
+
+    >>> import opendp.prelude as dp
+
+The methods of this module will then be accessible at ``dp.sklearn.linear_model``.    
+
+If you're interested in the underlying algorithm, we've also
+`implemented Theil-Sen Regression as a demonstration of OpenDP plugins <../user-guide/plugins/theil-sen-regression.html>`_.
+'''
+
+from typing import Iterable
+from opendp.extras.sklearn.linear_model._make_private_theil_sen import (
+    make_private_theil_sen as _make_private_theil_sen,
+)  # noqa: F401
+from opendp._lib import import_optional_dependency
+from opendp.mod import assert_features
+
+
+class LinearRegression:
+    """
+    DP Linear Regression
+
+    The interface is parallel to that offered by sklearn's
+    `LinearRegression <https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html>`_.
+    The ``fit`` method returns an sklearn ``LinearRegression`` object.
+
+    :param fraction_bounds: Lower and upper bounds x_bounds to retain when ``fit`` is called.
+    """
+    def __init__(self, fraction_bounds: tuple[float, float] = (0.25, 0.75)):
+        self.fraction_bounds = fraction_bounds
+
+    def fit(
+        self,
+        X,
+        y,
+        x_bounds: Iterable[tuple[float, float]],
+        y_bounds: tuple[float, float],
+        scale: float,
+        runs: int = 1,
+        candidates_count: int = 100,
+    ):
+        """
+        Fit DP linear model.
+
+        Required features: ``floating-point``
+
+        :param X: Training data. Array-like of shape (n_samples, 1)
+        :param y: Target values. Array-like of shape (n_samples,)
+        :param x_bounds: Bounds for training data; For the moment, only lists containing a single tuple are supported
+        :param y_bounds: Bounds for target data
+        :param scale: The scale of the noise to be added
+        :param runs: Controls how many times randomized pairwise predictions are computed. Increasing this value can improve the robustness and accuracy of the results; However, it can also increase computational cost and amount of noise needed later in the algorithm.
+        :param candidates_count: How many evenly spaced candidates to generate
+        :return: A fitted sklearn ``LinearRegression``
+
+        :example:
+
+        >>> import opendp.prelude as dp
+        >>> try:
+        ...    import sklearn
+        ... except ModuleNotFoundError:
+        ...     import pytest
+        ...     pytest.skip('Requires extra install')
+        >>> dp.enable_features("floating-point")
+        >>> lin_reg = dp.sklearn.linear_model.LinearRegression().fit(
+        ...     X=[[1], [2], [3], [4], [5]],
+        ...     y=[1, 2, 3, 4, 5],
+        ...     x_bounds=[(0, 10)],
+        ...     y_bounds=(0, 10),
+        ...     scale=1,
+        ... )
+        >>> lin_reg.predict([[10]])
+        array([...])
+        """
+        assert_features("floating-point")
```

Comment:
If we assert a feature, it should be done in the relevant constructor(s), not in a wrapper API. The constructors should be correct independently of how they are wrapped. 

That being said, this doesn't suffer from floating-point issues, because the transformation is microdata -> microdata, the maps are exact, and the actual mechanism is hardened. So the assertion can be dropped.